src/pages/SelectLanguage/SelectLanguage.js

```diff
@@ -0,0 +1,96 @@
+import { useState, useEffect, React } from 'react'
+import { StyleSheet, Text, View } from 'react-native'
+import { Picker } from '@react-native-picker/picker'
+import { useSelector, useDispatch } from 'react-redux'
+
+import colors from '../../theme/colors'
+import languages from './languageList'
+import {
+  setInputLanguage,
+  setOutputLanguage,
+} from '../../slices/targetLanguageSlice'
+
+export default function SelectLanguage() {
+  const currInput = useSelector((state) => state.targetLanguage.input)
+  const currOutput = useSelector((state) => state.targetLanguage.output)
+  const [chosenInput, setChosenInput] = useState(currInput)
+  const [chosenOutput, setChosenOutput] = useState(currOutput)
+
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    dispatch(setInputLanguage(chosenInput))
+  }, [chosenInput])
+
+  useEffect(() => {
+    dispatch(setOutputLanguage(chosenOutput))
+  }, [chosenOutput])
+
```

Comment:
I'd have to play with it some more... but I don't think you necessarily need both the useEffects above and the value change handlers below. I think you'd be updating the state twice, and maybe make it an endless loop?... onValueChange is calling the handlers that are setting the React state... useEffect is watching to see if the React states are changing... then dispatching the change to the Redux state. But the useSelectors are watching to see if the redux state is changing and will update... but the valuechange handlers are also dispatching the redux state change.

I think this is most easily accomplished by updating the Redux state (this way you'll have access to the state in other components besides this language picker).
You can do this by removing the React states (lines 16 and 17) and their setStates in the onvaluechange handlers and only use the selectors and dispatch. The selectors will pull the initial values for currInput and currOutput from state and then "watch" the redux state to see if it changes. OnValueChange handlers will dispatch the change to the redux slice... update the redux slice with the appropriate value... then the selectors will see that the state changed and update your currInput and currOutput values for the user to see.